Update the Docs handbook for Weeks 1-2 of the release (#3017)
kubernetes/sig-release: release-team/role-handbooks/docs/Release-Timeline.md

Reviewer: ```suggestion
    - **No PR:** No PR is open (helpful for deadline tracking)
```

Reviewer: ```suggestion
- ⚠️ Make sure that every docs PRs for the release have the correct base and set the correct Milestone. For example, enhancement PRs for version 1.21 need a base branch of `dev-1.21` and the Milestone set to `1.21`.
```

Reviewer: ```suggestion
    - [Hold a docs-only retro with SIG Docs](#hold-a-docs-only-retro-with-sig-docs)
```

Reviewer: ```suggestion
Define deadlines clearly e.g. End of Day (EOD) PST is not clear if EOD is 5:00 PM or 11:59 PM PST.
```
There's not a time conversion issue here, but rather distinction between whether EOD corresponds to working hours (i.e. ending at 5PM) or actual day end (i.e. 11:59:59 PM)

Reviewer: Clarification: I've changed the `AM` --> `PM`
Apologies if it was confusing earlier 😓

- [Release Timeline](#release-timeline)
  - [Overview](#overview)
  - [Early Steps (Weeks 1-2)](#early-steps-weeks-1-2)
    - [Connect with Release Team](#connect-with-release-team)
      - [Getting Started](#getting-started)
      - [Slack & Messaging](#slack--messaging)
      - [Google Groups](#google-groups)
    - [Shadow Selection & Onboarding](#shadow-selection--onboarding)
      - [Select Shadows](#select-shadows)
      - [Contact Selected Shadows](#contact-selected-shadows)
    - [Shadow Onboarding Meeting](#shadow-onboarding-meeting)
      - [Enhancement Tracking Setup](#enhancement-tracking-setup)
    - [Update Release Team & Access Control](#update-release-team--access-control)
      - [Access & Permission Setup](#access--permission-setup)
    - [Introduce Yourself to Localization Teams](#introduce-yourself-to-localization-teams)

>     - [Responsibilities Sign up Sheet](#responsibilities-sign-up-sheet)

Reviewer: ```suggestion
    - [Responsibilities Sign-up Sheet](#responsibilities-sign-up-sheet)
```

#### Select Shadows

You will be provided with survey results from people interested in working on the release team. If you do not have the results, contact the Release Lead or the Subproject Leads for more information.

After vetting the volunteers for their roles, role leads should make a final decision on selected shadows with the incoming Release Team Lead & Release Team Subproject Lead. In the past, the SIG Docs release team has been between 3-6 members. Read the following document to know more about the role of [Shadow in the Release Team](https://github.com/kubernetes/sig-release/blob/master/release-team/release-team-selection.md#shadows).

Drop a Slack DM or email the incoming candidates for shadows to check if they are still interested and available:
```markdown
Hello! 👋

You have been selected to be a Kubernetes [future-release] Release Docs Shadow. Can you please confirm that you're still interested and available?

Thanks,
[Docs Lead Name]
```

Once all the shadows have been selected and confirmed, add the contacts to the Contact Sheet to be shared by the Release Lead / Subproject Lead. Example: [v1.21 Contact Sheet](https://bit.ly/k8s121-contacts)

> [!Important]

> > Beyond meeting the basic requirements and time commitments, a good Shadow is someone who is active in SIG-Docs. Selection priority should go to previous shadows who want to eventually lead a SIG-Docs release. Other than those few guidelines, use your best judgement!

Reviewer 2: ```suggestion
> Beyond meeting the basic requirements and time commitments, a good shadow is someone who is active in SIG Docs. Selection priority should go to previous shadows who want to eventually lead a SIG Docs release. Other than those few guidelines, use your best judgement!
```

---

### Introduce Yourself to Localization Teams

- Create GitHub discussion in [k/sig-release](https://github.com/kubernetes/sig-release/discussions/) to introduce yourself to the current localization owners and to sync up early on the release timeline.

**Template:**
```markdown
Title: K8s [future release] - Formalize docs release strategy with docs localization owners

Hello 👋,

I'm [Your Name], the SIG Release Docs lead for the [future release] release.

I'm opening this discussion to coordinate the docs release (tentatively scheduled for [Release Date]) with localization owners. I look forward to working with everyone! Let me know if you have any questions!

Thank you! 🙏

cc: @kubernetes/sig-docs-fr-owners @kubernetes/sig-docs-de-owners @kubernetes/sig-docs-hi-owners @kubernetes/sig-docs-id-owners
@kubernetes/sig-docs-ja-owners @kubernetes/sig-docs-ru-owners @kubernetes/sig-docs-pl-owners @kubernetes/sig-docs-ko-owners
@kubernetes/sig-docs-pt-owners @kubernetes/sig-docs-es-owners @kubernetes/sig-docs-zh-owners @kubernetes/sig-docs-vi-owners
@kubernetes/sig-docs-uk-owners @kubernetes/sig-docs-it-owners
```
---

### Responsibilities Sign up Sheet

> Create a multisheet spreadsheet with `Weekly Branch Sync`, `Release Notes PR Creation/Review`, `Release Team Updates`, `SIG-Docs (APAC) Updates`, `SIG-Docs (EMEA) Updates`, `Burndown Meeting` sheets. This will serve as an internal schedule and signup sheet for the Docs team to follow. The schedule is used to track progress and give status during the release team meetings.

Reviewer 2: ```suggestion
Create a multisheet spreadsheet with `Weekly Branch Sync`, `Release Notes PR Creation/Review`, `Release Team Updates`, `SIG Docs (APAC) Updates`, `SIG Docs (EMEA) Updates`, `Burndown Meeting` sheets. This will serve as an internal schedule and signup sheet for the Docs team to follow. The schedule is used to track progress and give status during the release team meetings.
```

- Track Doc PRs for KEPs in the [website](https://github.com/kubernetes/website) project, plus all Doc PRs for the dev-[future release] branch that may include general updates without a KEP.
- By default, all opted-in KEPs will `Need Docs`. Enhancement owners/contributors should confirm the enhancement does not need new or updated documentation.

> [!Note]

> > - The blog-PR  is tracked by the Release Comms Team.

Reviewer: ```suggestion
> - The blog-PR is tracked by the Release Comms Team.
```

> - The kubernetes/website repo changed from using a `master` branch to a `main` branch in 2021. 

> > Be aware that several linked, example PRs uses the `master` branch.

Reviewer: ```suggestion
> - The kubernetes/website repo changed from using a `master` branch to a `main` branch in 2021. Be aware that several linked, example PRs uses the `master` branch.
```

Actual file render doesn't show the `Be aware ...` line on newline

<img width="1291" height="520" alt="Image" src="https://github.com/user-attachments/assets/1883cb46-52db-4707-b5ab-3ba9c29b495b" />

> > Be aware that several linked, example PRs uses the `master` branch.

Reviewer: ```suggestion
> Be aware that several linked, example PRs use the `master` branch.
```

Author: The intended word here is actually `uses`

> [!Tip]
> For the end-of-year release, KubeCon NA and the US Thanksgiving holiday can be additional impediments to Docs-related deadlines. Please communicate to enhancement owners/contributors about upcoming deadlines earlier and more frequently.

---

## Early Steps (Weeks 1-2)

These steps take approximately one hour to complete, and should be completed immediately.

### Connect with Release Team

#### Getting Started

- [ ] Familiarize yourself with the [release team and process](https://github.com/kubernetes/sig-release/tree/master/release-team).

#### Slack & Messaging

> - [ ] Join `#sig-release` Slack channel and introduce yourself as Docs Lead for [future release].

Author: Note to self: Migrate the checkboxes into bullet. Checkboxes would makes sense in the tracking issue for Docs Lead.

@kernel-kun

| [kubernetes-sig-docs](https://groups.google.com/forum/#!forum/kubernetes-sig-docs) | No | Directly join the mailing list via the URL |
| [kubernetes-dev](https://groups.google.com/a/kubernetes.io/g/dev) | No | Directly join the mailing list via the URL |
| [kubernetes-release-team](https://groups.google.com/a/kubernetes.io/g/release-team) | **Yes** | List of members for this group is managed in git. Please create a PR against `kubernetes/k8s.io` repo to include your email under `release-team` group in [`sig-release/groups.yaml`](https://github.com/kubernetes/k8s.io/blob/main/groups/sig-release/groups.yaml) |
| [kubernetes-release-team-shadows](https://groups.google.com/a/kubernetes.io/g/release-team-shadows) | **Yes** | List of members for this group is managed in git. Please create a PR against `kubernetes/k8s.io` repo to include your email under `release-team-shadows` group in [`sig-release/groups.yaml`](https://github.com/kubernetes/k8s.io/blob/main/groups/sig-release/groups.yaml) |

---

### Shadow Selection & Onboarding

#### Select Shadows

> You will be provided with survey results from people interested working on the release team. If you do not have the results, contact the Release Lead or the Subproject Leads for more information.

Reviewer: ```suggestion
You will be provided with survey results from people interested in working on the release team. If you do not have the results, contact the Release Lead or the Subproject Leads for more information.
```

**Resources**:
- Know your Timelines: [Release Timeline](https://github.com/kubernetes/sig-release/blob/master/releases/release-[future-release]/README.md) to keep track of where we are in the cycle and what is expected at each interval, especially the Docs deadlines as they’ll be crucial for our team. I will also provide updates and reminders as we progress.

**Meetings**:
- Shadow Onboarding Meeting: I'd love to jump on a call to go over the release process with everybody, describe what we'll be doing, and answer any questions. Below is a poll to see what time works best for everyone. As we near the end of the release cycle I will add a regular weekly check-in. Please select your preferred time(s): [include the Doodle Link]
- Also @[Subproject Lead] will be hosting a general release shadow orientation soon - stay tuned!
```

---

### Shadow Onboarding Meeting

Schedule a shadow onboarding meeting with the incoming shadows. The meeting will help everyone get to know each other and break the ice, while also walking them through a high-level overview of the release process and what's coming in the next few months.

- **Duration:** About 30-60 minutes

> - **Record:** Yes, if posssible. Share with team for reference.

Reviewer: ```suggestion
- **Record:** Yes, if possible. Share with team for reference.
```

**Action Items:**
- [ ] Schedule meeting using poll results
- [ ] Create meeting agenda
- [ ] Record session (if possible).

> - [ ] Share recording and slides (if any) with team.

Reviewer: Should we go with Checkboxes of Bullets for action items?
cc. @Urvashi0109 
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/ac4bad58-72b8-43f0-94be-1189b2815c9c" />

**Agenda:**
1. Introductions (5-10 min)
1. Release process overview (15-20 min)
1. Walk through this timeline document (15-20 min)
1. Q&A (10-15 min)
1. Set expectation that the mantra is "Hurry up & Wait" - but it does get hectic in the last few weeks.

> ---

Reviewer: ```suggestion
```

Since the newly added rule separators from the top of document this till current heading is adding rulers only for `###` H3 headings, we can probably remove this one or probably re-evaluate whether newer ones need to be consistent for other sub-headings as well

> `---` locations : 111, 143, 218, 240, 256, 290, 313, 368, 406, 434 (base only had 1 at line 77).

#### Enhancement Tracking Setup

> Early in the release cycle, the Enhancements Lead opens a GitHub Project, e.g., [the 1.26 Enhancement Tracking](https://github.com/orgs/kubernetes/projects/98/views/1). Later in the release, this project board will contain important information for the Docs team. Check the Docs view [here](https://github.com/orgs/kubernetes/projects/98/views/6):

Reviewer: Should we add a comment on who to reach out to incase the Shadows or Lead don't have access to the Github Project Board
Similar to what is mentioned at [L272](https://github.com/kubernetes/sig-release/pull/3017/changes#diff-f0f313d56a6fbf990236fbf8f7c81d993c680a6c085420ae1e6d307bb1a86e38L272)

- Familiarize yourself with the Docs view of the tracker.
- Which enhancements the team will be tracking for this release.
- Feature owners (and their GitHub IDs).
- Links to docs PRs opened for each enhancement.
- Docs requirement status (default: "Needs Docs").
- PR documentation status (No PR, Draft PR, Ready-for-Review PR, Merged PR).

> [!Important]
> Until [1.25](https://tinyurl.com/k8s125-enhancements), a tracking spreadsheet was used. e.g.: [the 1.21 release spreadsheet](http://bit.ly/k8s121-enhancements).

---

### Update Release Team & Access Control

#### Access & Permission Setup

- [ ] Make sure you are included in the `k/sig-release` release team group [sig-release/group.yaml](https://github.com/kubernetes/k8s.io/blob/main/groups/sig-release/groups.yaml)
  - Example: [v1.35 PR to update release team groups](https://github.com/kubernetes/k8s.io/pull/8498)

> - [ ] Open a PR to add the Docs Team in the `release-[future release]`. Here is an example of the [v1.35 Release Team](https://github.com/Urvashi0109/sig-release/blob/558161ffcdd2791543a233bd19f88b9e62b06cd8/releases/release-1.35/release-team.md).

Reviewer: ```suggestion
- [ ] Open a PR to add the Docs Team in the `release-[future release]`. Here is an example of the [v1.35 Release Team](https://github.com/kubernetes/sig-release/blob/master/releases/release-1.35/release-team.md).
```

  - Example PR: [v1.35 Docs Shadows PR](https://github.com/kubernetes/sig-release/pull/2875)

> - [ ] Please create a PR against `k/sig-release` repo to include the email ids of the Docs Team member of this release under `release-team` and `release-team-shadows` group in [sig-release/group.yaml](https://github.com/kubernetes/k8s.io/blob/main/groups/sig-release/groups.yaml).

Reviewer: Should we revert it back to `kubernetes/sig-release` as L133 previously, or go with abbreviation?
```suggestion
- [ ] Please create a PR against `kubernetes/sig-release` repo to include the email ids of the Docs Team member of this release under `release-team` and `release-team-shadows` group in [sig-release/group.yaml](https://github.com/kubernetes/k8s.io/blob/main/groups/sig-release/groups.yaml).
```

  - Example: [v1.35 Docs Shadows PR](https://github.com/kubernetes/k8s.io/pull/8585)

> - [ ] Open a PR to add yourself (Docs Lead Only) to `k/k` [CHANGELOG OWNERS](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/OWNERS) `approvers` and `reviewers` section.

Reviewer: Should we revert it back to `kubernetes/kubernetes` as in L135 previously, or it's better to go with `k/k`?
```suggestion
- [ ] Open a PR to add yourself (Docs Lead Only) to `kubernetes/kubernetes` [CHANGELOG OWNERS](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/OWNERS) `approvers` and `reviewers` section.
```

Author: Explicit is better than implicit :) This would be referenced by shadows and clarity is of most importance

⚠️ Please coordinate with Emeritus Adviser for the release before taking this step.
Send a Slack message to those that you didn't select, e.g:
> [!CAUTION]
> **The next should be done by the Docs Lead**

> Hey, I'm Jim Angel (Docs Lead for SIG Docs). Thanks for your interest in the SIG Docs 1.14 release team!
>
> The release team for sig-doc shadows has no additional availability, but please stick around help out with some of our other sigs (including sig-docs)!
>
> How can you help?
> - Attend the sig release meetings (07 30 AM PST Bi-weekly on Tuesdays - see #sig-release for more info).
> - SIG-DOCs is always looking for new contributors, please go introduce yourself and we're happy to help! As a bonus, you will be preferred during the next release cycle as opposed to someone not involved with sig-docs.
     - Slack is full of other great SIGs that could always use your help!
>
> I am also a resource to reach out to if you have any community questions (there's also #sig-contribex). I started out as a volunteer and now I am part of many Kubernetes sigs and teams.
>
> Thanks again for your interest and time!
>

> > Jim Angel

Reviewer: This template is gonna be completely removed from handbook, it's fine if it's intentional

Author: Yes, The rejection I believe is now handled by the subproject owners - so we can skip it.

> > [!IMPORTANT]

Reviewer: ```suggestion
> [!Important]
```
To be consistent with other callouts

> Example: [v1.35 Doc Signup](https://docs.google.com/spreadsheets/d/1bij8ayB_yf8mPic1upcf13N5yHRF7HguIoYXLvU1Mss/edit?usp=sharing) spreadsheet

Reviewer: This spreadsheet opens up with default permissions as Edit which could let bad actors mangle the reference information
We should probably replace it with a xlsx published file link instead:

<img width="714" height="758" alt="Image" src="https://github.com/user-attachments/assets/623501f4-81d3-4e29-8403-eb9284f5332f" />

<img width="1304" height="873" alt="Image" src="https://github.com/user-attachments/assets/a92b20ea-88c0-49bb-96c0-c00edbe9bafc" />

Reviewer 3: fyi, if you are interested, you can avoid linking to a google doc here, by checking in a markdown file instead.

this is what we have for enhancements handbook:
- [checked in template](https://github.com/kubernetes/sig-release/blob/master/release-team/role-handbooks/enhancements/meeting-rotation-template.md)
- [instructions to copy the template](https://github.com/kubernetes/sig-release/tree/master/release-team/role-handbooks/enhancements#meeting-update-rotation)

i don't know if you can copy/paste it into a 'google sheet' but it works for pasting it as a table in a google doc

Reviewer 3: [here](https://github.com/kubernetes/sig-release/blob/master/release-team/role-handbooks/enhancements/sig-outreach-assignment-template.md?plain=1) is another template that uses just markdown, no html

Author: Thanks @jenshu for the template - I think that's a good approach here as well. I will update the PR.

>   Weekly Branch Sync is typically done one a Friday.

Reviewer: ```suggestion
  Weekly Branch Sync is typically done on a Friday.
```

>   * Column A `Date`: Include all the dates of Friday's until the Release

Reviewer: ```suggestion
  * Column A `Date`: Include all the dates of Fridays until the Release
```

>   This sheet is used to assign the responsibility among the team to provide updates to the SIG Docs (EMEA) Meetings

Reviewer: ```suggestion
  This sheet is used to assign the responsibility among the team to provide updates to the SIG Docs (EMEA) Meetings.
```

>   This sheet is used to assign the responsibility among the team to provide updates to the SIG Docs (APAC) Meetings

Reviewer: ```suggestion
  This sheet is used to assign the responsibility among the team to provide updates to the SIG Docs (APAC) Meetings.
```

```
Now create a pull request that targets the future release (example: `dev-1.21`) **not** `main`.

---

### Attend Meetings & Slack Updates

> [!Important]
> - Attend the release team meeting outlined in the current release. If you cannot attend, a Shadow needs to attend in your place.
> - Attend [SIG Docs meetings](https://github.com/kubernetes/community/tree/master/sig-docs#meetings) to report a status update. 
> If you cannot attend, a Shadow needs to attend in your place and/or update the [SIG Docs meeting notes](https://docs.google.com/document/d/1emuO4nmaQq3K8JZ9-MQeIygtrCPO9kWv7U7RzTaW4F8/edit?usp=sharing) with the information necessary to keep the community informed (e.g. staffing updates, branch syncs, placeholder PR deadlines, etc).
> - The Docs Lead and Shadows must either attend all release team and SIG Docs meetings or communicate with the team if unable to attend. Every team member should have equal opportunities to provide verbal status updates in the meetings. The Docs Lead must ensure Shadows are well-informed on the status updates that release team and SIG Docs need, while Shadows must actively monitor progress to report on.

> > [!NOTE]

Reviewer: ```suggestion
> [!Note]
```
To be consistent with other callouts

> > SIG Docs prefers and welcomes a status of yellow or red on anything that needs attention, has the possiblity to miss deadlines or target dates, or requires follow-up from feature authors and SIGs that you're not receiving.

Reviewer: ```suggestion
> SIG Docs prefers and welcomes a status of yellow or red on anything that needs attention, has the possibility to miss deadlines or target dates, or requires follow-up from feature authors and SIGs that you're not receiving.
```
